### PR TITLE
[0.65] Cherry-pick Playground Cert Removal PRs

### PR DESCRIPTION
**Pull request by NickGerleman**
The playground temp certificate expired, failing the PR pipeline in stable branches. This brings back changes to not rely on playground certificate.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8122)

### CODE REVIEW

File: .ado/jobs/playground.yml
Comment (dannyvv) [inline .ado/jobs/playground.yml:71]: @chiaramooney : Just to double-check: I thought you created a shared helper yaml for this scenario?
@NickGerleman : I'll leave it up to you to backport that helper and use it, or go with manual expansion here.
Note: Just mentioning to share awareness.

File: .ado/jobs/playground.yml
Comment (dannyvv) [inline .ado/jobs/playground.yml:140]: This looks pretty useful, should we file an issue to forward port this to main branch if not there?

**Review by dannyvv — APPROVED** [review]

File: .ado/jobs/playground.yml
Comment (NickGerleman) [reply .ado/jobs/playground.yml:140]: It should be there, since this was a straight cherry-pick of relevant PRs.

File: .ado/jobs/playground.yml
Comment (NickGerleman) [reply .ado/jobs/playground.yml:71]: It's also very possible I missed a commit. I tried to find the original one, and any followup fixes.

**Comment by NickGerleman** [comment]
Merging to unblock CI

**Pull request merged**
